lms/tasks/email_digests.py

```diff
@@ -66,9 +72,27 @@ def send_instructor_email_digest_tasks(*, batch_size):
                 )
             ).cte("candidate_courses")
 
-            h_userids = request.db.scalars(
-                select(User.h_userid)
-                .select_from(User)
+            tasks_done_subsubquery = (
+                select(
+                    TaskDone.data,
+                    func.rank()  # pylint:disable=not-callable
+                    .over(
+                        order_by=TaskDone.data["created_before"].desc(),
+                        partition_by=TaskDone.data["h_userid"],
+                    )
+                    .label("rank"),
+                ).where(TaskDone.data["h_userid"].as_string() == User.h_userid)
+            ).subquery()
```

Comment:
I'm using `subquery()` but we may want to use `cte()` as is done with `candidate_courses` above, I'm not sure what the difference is